euphonic/design/06_powder_convergence_discussion.md

```diff
@@ -0,0 +1,159 @@
+     sampling more q-points.
+
+   - At intermediate range, the error-vs-|q| trend can be somewhat flattened by allowing
+     npts to follow |q|^2; i.e. sampling at a consistent _area density_.
+
+
+## Suggested strategy
+
+   - A multi-segment scheme in which minimum/maximum npts are set, and
+     in-between npts is a multiple of |q|^2 (where |q| is somehow
+     normalised to the reciprocal lattice).
+   - This model has three parameters (a bit high) but they are all easily understood:
+     - minimum qpts (used near the Gamma-point)
+     - maximum qpts (used at long-range)
+     - qpts density at sphere surface (in qpts Å^2 or, inequivalently
+       but perhaps more intuitively, in qpts / (1/Å radius sphere))
+
+   - Consider an analytic or simplified method for long-range where
+     sphere can be considered good average of BZ.
+
+     - DV raised that this may not be consistent with the generally
+       rigorous approach taken to accuracy in this project; general
+       agreement that such assumptions should be theoretically
+       justified and backed-up empirically.
+
+     - Another cost-saver would be the use of Brille to map qpts to BZ
+       and interpolate the spectrum.
+       - This requires _periodic_ contributions to spectrum to be
+         separated from terms that depend on absolute _q_.
+       - Hopefully that is just the Debye-Waller term and it will work.
+
+# Incidental observations
+  - The calculations were quite slow to run (even with
+    multithreading), which could negatively impact the end-user
+    experience.
+
+    - It has not yet been determined whether this was due to poor
+      optimisation of the coherent sphere-averaging tools or is a
+      limitation in the underlying spectrum calculation
+
+    - One severe issue that can be addressed was memory usage when
+      computing large numbers of qpts; this could be reduced by some
+      kind of serial batching/chunking to reduce the size of arrays.
+
+# Summary of decisions
+  - A basic implementation of numerical powder-averaging should be
+    included in the upcoming research paper and corresponding Euphonic
+    release.
+  - Investigation of smarter approaches and Brille integration will need to wait.
+
+# Actions
+
+## High-priority
+   - Profile powder-averaging script to identify sources of
+     poor performance (AJJ & RF)
+   - Examine impact of jitter in r-direction; does this clear up outliers? (AJJ)
+   - Implement user API and scripts to make use of proposed 3-parameter powder-averaging scheme.
+     - Try min/max instead of geometric mean
+
+## Longer-term
+   - Look into Sobel sampling (AJJ)
+   - Consider powder-averaging at large q as a good
+     application/demonstration of Brille-Euphonic interface
+     (Euphonic-Brille team)
+   - Look into fully volumetric sampling (AJJ)
+   - Consider interactions between powder-averaging and instrumental
+     resolution functions (?)
```

Comment:
I'd sic Toby on this one!

Reply:
Done! (In these notes.) I have also added a note to https://github.com/pace-neutrons/Pace-Project-Plan/issues/116 which isn't assigned to anyone.